bundles/org.openhab.binding.irtrans/src/main/java/org/openhab/binding/irtrans/internal/handler/EthernetBridgeHandler.java

```diff
@@ -439,13 +442,13 @@ private String[] getRemoteCommands(String remote, int index) {
         String[] commandList = new String[0];
 
         if (response != null) {
-            String message = stripByteCount(response).split("\0")[0];
-            logger.trace("commands returned {}", message);
-            if (message != null) {
-                if (message.contains("COMMANDLIST")) {
-                    commandList = message.split(",");
+            String[] messages = stripByteCount(response).split("\0");
+            logger.trace("commands returned {}", messages[0]);
```

Comment:
I suppose this logging will still cause `ArrayIndexOutOfBoundsException` for empty array, since `length` is only checked in next line?

Reply:
Double checked and `ArrayIndexOutOfBoundsException` can't happen with java.string.Split so no need to check for length as the array will allways hold 1 value (the original string if the split delemiter was not found)

I updated the fix.

Reply:
Was partly fixed, should not code anything when to tired.. i corrected it. should be good now.